tests/websockets/src/jvmTest/kotlin/incubating/RetryTest.kt

```diff
@@ -0,0 +1,100 @@
+package incubating
+
+import app.cash.turbine.test
+import com.apollographql.apollo.sample.server.SampleServer
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.exception.ApolloHttpException
+import com.apollographql.apollo3.exception.ApolloNetworkException
+import com.apollographql.apollo3.interceptor.RetryOnErrorInterceptor
+import com.apollographql.apollo3.mockserver.MockResponse
+import com.apollographql.apollo3.network.ws.incubating.SubscriptionWsProtocol
+import com.apollographql.apollo3.network.ws.incubating.WebSocketNetworkTransport
+import com.apollographql.apollo3.testing.internal.runTest
+import com.apollographql.apollo3.testing.mockServerTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Test
+import sample.server.TagSubscription
+import sample.server.ZeroQuery
+import kotlin.test.assertIs
+
+class RetryTest {
+  @Test
+  fun retryIsWorking() = runTest(skipDelays = false) {
+    // Hopefully it's really a free port, if not this test will fail
+    val freePort = 8392
+    var sampleServer = SampleServer(freePort, "tag1")
```

Comment:
We could extract the port from sampleServer.subscriptionsUrl(). But the hardcoded port is most probably fine.

Reply:
Excellent point, done in https://github.com/apollographql/apollo-kotlin/pull/5685/commits/297950d86f4e5b21421b323b9a9d4bce26f24e9b